### PR TITLE
Improve validation for Event' location

### LINKED ISSUE #197: [PE-D] Invalid location for creating new event not caught
Invalid location is not caught.

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: jianhandev/ped#2

### LINKED ISSUE #198: [PE-D] Empty string location
Location is an empty string, but event still got added. 
UG should be clearer on the constraints of the location parameter.

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: jianhandev/ped#3

### PR DESCRIPTION
**Pull request by jieffyy**
- Improve location's validation.
Closes #197, closes #198.

- Small change to `.isSameEvent()`, as part of #208.

### CODE REVIEW

File: src/main/java/seedu/address/model/event/EventName.java [hunk 1]
```diff
@@ -9,13 +9,14 @@
  */
 public class EventName {
 
-    public static final String MESSAGE_CONSTRAINTS = "Event names can take any values, and it should not be blank";
+    public static final String MESSAGE_CONSTRAINTS =
+            "Locations should only contain alphanumeric characters and spaces, and it should not be blank";
```
Comment (teekoksiang) [inline src/main/java/seedu/address/model/event/EventName.java]: `"Locations"` is `Event's name` here?

File: src/main/java/seedu/address/model/event/Event.java [hunk 2]
```diff
@@ -75,7 +75,6 @@ public boolean isSameEvent(Event otherEvent) {
 
         return otherEvent != null
                 && otherEvent.getName().equals(getName())
-                && otherEvent.getDescription().equals(getDescription())
                 && otherEvent.getLocation().equals(getLocation())
                 && otherEvent.getEventDate().equals(getEventDate());
```
Comment (teekoksiang) [inline src/main/java/seedu/address/model/event/Event.java:79]: One thing to consider, when comparing `name` and `location`, should it use `equalsIgnoreCase`?

File: src/test/java/seedu/address/model/event/LocationTest.java [hunk 3]
```diff
@@ -22,39 +22,44 @@ void constructor_null_throwsNullPointerException() {
...
     void constructor_validLocation_success() {
-        assertNotNull(new Location(LOCATION_WITH_CAROT));
-        assertNotNull(new Location(LOCATION_WITH_AT));
-        assertNotNull(new Location(LOCATION_WITH_SLASH));
-        assertNotNull(new Location(LOCATION_WITH_TRAILING_SPACES));
+        assertNotNull(new Location(VALID_LOCATION));
     }
 
     @Test
     void isValidLocation() {
```
Comment (teekoksiang) [inline src/test/java/seedu/address/model/event/LocationTest.java:46]: Should there be some positive test cases as well?
Similar for `EventNameTest`.

**Review by teekoksiang — COMMENTED** [review]
Almost LGTM, some minor nits to consider.

File: src/main/java/seedu/address/model/event/EventName.java (hunk 1 above)
Comment (jieffyy) [reply src/main/java/seedu/address/model/event/EventName.java]: Good catch, fixed.

File: src/test/java/seedu/address/model/event/LocationTest.java (hunk 3 above)
Comment (jieffyy) [reply src/test/java/seedu/address/model/event/LocationTest.java:46]: Good point. Fixed.

File: src/main/java/seedu/address/model/event/Event.java (hunk 2 above)
Comment (jieffyy) [reply src/main/java/seedu/address/model/event/Event.java:79]: Good point - fixed.

**Review by teekoksiang — APPROVED** [review]
Good job! LGTM.

**Pull request merged**